### PR TITLE
Added headers interceptor to grpc transport

### PR DESCRIPTION
**Pull request by chamorin**
## PR-Codex overview

> The following files were skipped due to too many changes: `package-lock.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by ghost** [comment]
### Contributors

<a href="https://github.com/chamorin">chamorin</a>, <a href="https://github.com/DenisCarriere">DenisCarriere</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

**Comment by DenisCarriere** [comment]
`createHeadersInterceptor` method would be better placed `@substreams/core`, that way it can be used for Web Client side libraries to also push headers, the `node` library is for NodeJS

Also, for `State`, looks like it's been updated to `Progress`

Whenever you set a type as `any`, it usually means it should be either removed or refactored into something else

File: src/BlockEmitter.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { type CallOptions, type Transport, createPromiseClient } from "@bufbuild/connect";
 import { AnyMessage, IMessageTypeRegistry, JsonObject, Message } from "@bufbuild/protobuf";
-import { State, createStateTracker, isEmptyMessage, unpackMapOutput } from "@substreams/core";
```
Comment (DenisCarriere) [inline src/BlockEmitter.ts:3]: Changed to `Progress`

File: src/createDefaultTransport.ts [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
 import type { IMessageTypeRegistry } from "@bufbuild/protobuf";
 import { createAuthInterceptor } from "@substreams/core";
 
 export function createDefaultTransport(
   baseUrl: string,
-  token?: string,
-  typeRegistry?: IMessageTypeRegistry,
+  token: string,
+  registry: IMessageTypeRegistry,
+  headers?: any,
```
Comment (DenisCarriere) [inline src/createDefaultTransport.ts:11]: Use `Headers` instead of `any`

**Review by DenisCarriere — CHANGES_REQUESTED** [review]

File: src/createHeadersInterceptor.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import type { Interceptor } from "@bufbuild/connect";
+
+export function createHeadersInterceptor(headers: any): Interceptor {
```
Comment (DenisCarriere) [inline src/createHeadersInterceptor.ts:3]: Use `Headers` instead of `any`

Comment (DenisCarriere) [inline src/createHeadersInterceptor.ts:3]: Move method to `@substreams/core`

Perhaps in `/auth` folder? https://github.com/substreams-js/substreams-js/tree/main/packages/core/src/auth

Or `/utils`

https://github.com/substreams-js/substreams-js/tree/main/packages/core/src/utils

@fubhy can chime in for a preference

File: package.json
Comment (DenisCarriere) [inline package.json:1]: Need to version bump `@substreams/core`

File: src/createHeadersInterceptor.ts (hunk 3 above)
Comment (DenisCarriere) [inline src/createHeadersInterceptor.ts:3]: Looks like if we implement `createHeaderInterceptor`

We could refactor `createAuthInterceptor` to use this method instead.

https://github.com/substreams-js/substreams-js/blob/main/packages/core/src/auth/create-auth-interceptor.ts

**Pull request merged**